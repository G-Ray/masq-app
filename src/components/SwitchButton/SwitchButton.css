.SwitchButton {
	display: flex;
	align-items: center;
}

.switch-container {
	position: relative;
	width: 50px;
}

input[type="checkbox"] {
	z-index: 1;
}

input[type="checkbox"].switch  {
	position: absolute;
	opacity: 0;
	width: 100%;
	height: 100%;
}

input[type="checkbox"].switch + div {
	vertical-align: middle;
	width: 40px;
	height: 20px;
	border-radius: 999px;
  background-color: grey;
}

input[type="checkbox"].switch:checked + div {
	width: 50px;
	background-position: 0 0;
}

input[type="checkbox"].switch + div {
	width: 50px;
	height: 24px;
}

input[type="checkbox"].switch + div > div {
	float: left;
	width: 20px;
	height: 20px;
	border-radius: inherit;
	background: white;
	-webkit-transition-timing-function: cubic-bezier(1,0,0,1);
	transition-timing-function: cubic-bezier(1,0,0,1);
	-webkit-transition-duration: 0.4s;
	transition-duration: 0.4s;
	-webkit-transition-property: transform, background-color;
	transition-property: transform, background-color;
	margin-top: 2px;
	margin-left: 2px;
}

input[type="checkbox"].switch:checked + div > div {
	-webkit-transform: translate3d(26px, 0, 0);
	transform: translate3d(26px, 0, 0);
}

label {
	margin:auto;
	margin-right: 8px;
	color: #191919;
	font-size: 10px;
	font-weight: 600;
}
