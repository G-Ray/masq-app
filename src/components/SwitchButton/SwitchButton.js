import React from 'react'
import PropTypes from 'prop-types'

import './SwitchButton.css'

export default class SwitchButton extends React.Component {
  constructor (props) {
    super(props)
    this.state = { checked: this.props.checked || false }
    this.toggle = this.toggle.bind(this)
  }

  toggle () {
    this.setState({ checked: !this.state.checked })
    if (this.props.onClick) this.props.onClick()
  }

  render () {
    const { secondary, color } = this.props
    const label = this.state.checked ? 'ENABLED' : 'DISABLED'
    const translation = secondary ? '32px' : '26px'
    const cursorStyle = this.state.checked ? {
      transform: 'translate3d(' + translation + ', 0, 0)'
    } : {}
    const backgroundStyle = this.state.checked ? {
      backgroundColor: color
    } : {}

    return (
      <div className='SwitchButton'>
        {!this.props.secondary ? (<label>{label}</label>) : null}
        <div id='switch-container' className={secondary ? 'secondary' : ''} onClick={this.toggle}>
          <div id='background' className={secondary ? 'secondary' : ''} style={backgroundStyle} />
          <div id='cursor' className={secondary ? 'secondary' : ''}
            style={cursorStyle}
          />
        </div>
      </div>
    )
  }
}

SwitchButton.propTypes = {
  onClick: PropTypes.func,
  secondary: PropTypes.bool,
  checked: PropTypes.bool.isRequired,
  color: PropTypes.string.isRequired
}
