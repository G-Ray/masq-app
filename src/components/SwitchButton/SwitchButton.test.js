import React from 'react'
import renderer from 'react-test-renderer'
import SwitchButton from './SwitchButton'

it('renders SwitchButton primary unchecked', () => {
  const component = renderer.create(<SwitchButton checked={false} color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders SwitchButton primary checked', () => {
  const component = renderer.create(<SwitchButton checked color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders SwitchButton secondary unchecked', () => {
  const component = renderer.create(<SwitchButton secondary checked={false} color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders SwitchButton secondary checked', () => {
  const component = renderer.create(<SwitchButton secondary checked color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
