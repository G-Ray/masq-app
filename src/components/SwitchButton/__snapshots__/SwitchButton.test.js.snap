// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SwitchButton primary checked 1`] = `
<div
  className="SwitchButton"
>
  <label
    style={
      Object {
        "color": "blue",
      }
    }
  >
    ENABLED
  </label>
  <div
    className=""
    id="switch-container"
    onClick={[Function]}
  >
    <div
      className=""
      id="background"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
    <div
      className=""
      id="cursor"
      style={
        Object {
          "transform": "translate3d(26px, 0, 0)",
        }
      }
    />
  </div>
</div>
`;

exports[`renders SwitchButton primary unchecked 1`] = `
<div
  className="SwitchButton"
>
  <label
    style={
      Object {
        "color": null,
      }
    }
  >
    DISABLED
  </label>
  <div
    className=""
    id="switch-container"
    onClick={[Function]}
  >
    <div
      className=""
      id="background"
      style={Object {}}
    />
    <div
      className=""
      id="cursor"
      style={Object {}}
    />
  </div>
</div>
`;

exports[`renders SwitchButton secondary checked 1`] = `
<div
  className="SwitchButton"
>
  <div
    className="secondary"
    id="switch-container"
    onClick={[Function]}
  >
    <div
      className="secondary"
      id="background"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
    <div
      className="secondary"
      id="cursor"
      style={
        Object {
          "transform": "translate3d(32px, 0, 0)",
        }
      }
    />
  </div>
</div>
`;

exports[`renders SwitchButton secondary unchecked 1`] = `
<div
  className="SwitchButton"
>
  <div
    className="secondary"
    id="switch-container"
    onClick={[Function]}
  >
    <div
      className="secondary"
      id="background"
      style={Object {}}
    />
    <div
      className="secondary"
      id="cursor"
      style={Object {}}
    />
  </div>
</div>
`;
